tests/Tests/100__deploy/100__installation/102__post_install.robot

```diff
@@ -315,3 +317,20 @@ Replace "Prometheus" With "Grafana" In Rhods-Monitor-Federation
     OpenShiftLibrary.Oc Patch    kind=ServiceMonitor
     ...                   src={"spec":{"selector":{"matchLabels": {"app":"grafana"}}}}
     ...                   name=rhods-monitor-federation   namespace=redhat-ods-monitoring  type=merge
+
+Verify CPU And Memory Requests For Image And Base Cuda Pods
+    [Documentation]     Verifies CPU and Memory Requests for image and base cuda pods
+    [Arguments]   ${namespace}=redhat-ods-applications
+    ${pods} =  Oc Get  kind=Pod  namespace=${namespace}
+    FOR    ${pod}    IN  @{pods}
+        Run Keyword If    "cuda-s2i" in "${pod['metadata']['name']}"
+        ...    Verify Requests Contains Expected Values  cpu=2  memory=4Gi  requests=${pod['spec']['containers'][0]['resources']['requests']}
+        Run Keyword If    "minimal-gpu" in "${pod['metadata']['name']}" or "pytorch" in "${pod['metadata']['name']}" or "tensorflow" in "${pod['metadata']['name']}"
+        ...    Verify Requests Contains Expected Values  cpu=4  memory=8Gi  requests=${pod['spec']['containers'][0]['resources']['requests']}
+    END
+
+Verify Requests Contains Expected Values
+    [Documentation]     Verifies cpu and memory requests contain expected values
+    [Arguments]   ${cpu}  ${memory}  ${requests}
+    Should Be Equal As Strings    ${requests['cpu']}  ${cpu}
+    Should Be Equal As Strings    ${requests['memory']}  ${memory}
```

Comment:
Couldn't you reuse the `Verify CPU And Memory Requests And Limits Are Defined For All Containers In All Pods in Project` keyword here?
it loops over a list of pods and checks that limits and requests are defined for both cpu and memory if I understand correctly

Reply:
Updated